src/components/apps/launch/params/FileInput.js

```diff
@@ -9,14 +9,23 @@ import InputSelector from "../InputSelector";
 
 import ResourceTypes from "components/models/ResourceTypes";
 
+import { useBootstrapInfo } from "contexts/bootstrap";
+
 export default function FileInput({ param, ...props }) {
+    const bootstrapInfo = useBootstrapInfo()[0];
+    const pref = bootstrapInfo?.preferences;
     return (
         <InputSelector
             margin="normal"
             acceptedType={ResourceTypes.FILE}
             label={param?.label}
             helperText={param?.description}
             required={param?.required}
+            startingPath={
+                pref?.rememberLastPath
+                    ? pref?.lastFolder
+                    : pref?.defaultFileSelectorPath
+            }
```

Comment:
This isn't needed here, since the `InputSelector` is already using logic like this in its call to `useSelectorDefaultFolderPath`.

Reply:
Oops! Missed that. I guess I tested it without checking if the actual preference was set to remember last path 🤔 .